sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaSource.java

```diff
@@ -293,174 +287,44 @@ public boolean start() throws IOException {
       tikaMetadata = spec.getInputMetadata() != null ? spec.getInputMetadata()
           : new org.apache.tika.metadata.Metadata();
 
-      if (spec.getMinimumTextLength() != null) {
-        tikaHandler.setMinTextLength(spec.getMinimumTextLength());
-      }
-
-      if (!Boolean.TRUE.equals(spec.getParseSynchronously())) {
-        // Try to parse the file on the executor thread to make the best effort
-        // at letting the pipeline thread advancing over the file content
-        // without immediately parsing all of it
-        execService = Executors.newFixedThreadPool(1);
-        execService.submit(new Runnable() {
-          public void run() {
-            try {
-              parser.parse(is, tikaHandler, tikaMetadata, context);
-              is.close();
-            } catch (Exception ex) {
-              tikaHandler.setParseException(ex);
-            }
-          }
-        });
-      } else {
-        // Some parsers might not be able to report the content in chunks.
-        // It does not make sense to create extra threads in such cases
-        try {
-          parser.parse(is, tikaHandler, tikaMetadata, context);
-        } catch (Exception ex) {
-          throw new IOException(ex);
-        } finally {
-          is.close();
-        }
+      try {
+        parser.parse(is, tikaHandler, tikaMetadata, context);
+      } catch (Exception ex) {
+        throw new IOException(ex);
+      } finally {
+        is.close();
       }
-      return advanceToNext();
+      current = tikaHandler.toString().trim();
```

Comment:
If tikaHandler decided to generate a string with some leading/trailing whitespace, then according to this handler the whitespace is probably significant. The user can trim the string manually if they would like to - it doesn't feel like responsibility of TikaIO.

Reply:
This is the only comment I'm not comfortable with - I can not imagine, ever, spaces before or at the end of some meaningful content, be of any significance, given that we are talking now about TikaIO helping to deal with practical cases. Tika parsers only report whatever they can, if it is some presentation with N leading or trailing spaces, then the parsers will report them; you are right users can strip them, but that is why we are working on TikaIO - let users do less when it comes to processing the content...

Reply:
If they were insignificant, why did the parser return them? If "the parser doesn't trim the string" should be considered a bug in the parser, then I don't think it's TikaIO's job to fix the bug; if it shouldn't be considered a bug in the parser and rather part of its intended behavior, then it's not TikaIO's job to change the intended behavior of the parser the user asked for either. I think it's better to have TikaIO do one job - "glue Beam to the Tika library", rather than two jobs - "glue Beam to the Tika library and do minor fixups on Tika's output".

Let's approach it from this viewpoint:
* Might a user ever file a bug "I need TikaIO to trim the string returned by a parser, but it didn't"?
* Might a user ever file a bug "I need TikaIO to return the string returned by the parser verbatim, but it trimmed the string"?

I think bullet 1 is extremely unlikely, because, if for the user's purposes it _really_ matters that the string must be trimmed (I suspect in most cases it doesn't matter at all, e.g. if they're splitting the string into words for text indexing, it doesn't matter if it's trimmed or not), all the user needs to do is type the 6 characters `.trim()` wherever they use `parseResult.getContent()`.

Bullet 2 is also quite unlikely, but possible: whitespace can be significant in some documents. E.g. imagine that the user is ingesting Word documents with Shakespeare plays and converting them to text files. The Word documents may be indented using whitespace, such as here http://www.textfiles.com/etext/AUTHORS/SHAKESPEARE/shakespeare-romeo-48.txt - it starts with a bunch of whitespace. Trimming the string is unrecoverably losing the indentation.

Reply:
Sure referring to the Shakespeare play is most convincing, so I'll drop 'trim()' :-)